enhancements/olm-deletes-operands.md

```diff
@@ -0,0 +1,294 @@
+
+For this OLM should always overwrite and disable the CSV cleanup option to `spec.cleanup.enabled: false` regardless of what was configured in CSV initially packaged in the operator bundle.
+This can be done by overwriting the CSV during its initial `None` phase when it's first created, or alternatively when the CSV is first written out to the `InstallPlan` by the catalog operator.
+
+#### Opt-in only if cleanup is supported
+
+Since operator users can switch on the cleanup behavior for operators that don't actually implement Finalizers, OLM can end up deleting CRs for which there remain leftover secondary resources (off-cluster and cross-namespaced) that are now potentially untraceable because the root CR is deleted.
+
+To prevent this there should be a second field `spec.cleanup.supported` (default `false`).
+
+This field should be set to `true` by the operator/CSV author to convey if the operator has the capability to clean up all of its created resources via Finalizers.
+This field can also be `true` even if the operator has no finalizers but it is sufficient to rely on the garbage collector to cleanup on-cluster secondary resources on CR deletion.
+
+```YAML
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+...
+spec:
+  ...
+  cleanup:
+    supported: false
+    enabled: true
+```
+
+If cleanup is not supported by the operator `spec.cleanup.supported: false`, OLM will ignore the field `spec.cleanup.enabled` and default to no cleanup and not delete CRs.
+Only if `spec.cleanup.supported: true` will OLM act on `spec.cleanup.enabled` to enable/disable cleanup.
+
+This allows the operator author to signal cleanup capabilities while still letting the operator user be the one to explicitly opt-in.
+
+#### Override and force cleanup
+
+If an operator does not explicitly support cleanup, `spec.cleanup.supported: false`, but a user still wants OLM to initiate cleanup on operator uninstall, then OLM will require additional confirmation from the user. This ensures that OLM does not automatically take any destructive actions and that the user is aware of the operator's cleanup capability.
+
+When a user enables cleanup `spec.cleanup.enabled: true` while cleanup is unsupported `spec.cleanup.supported: false`, OLM will generate a random key string in the status block `status.cleanup.confirmationKey`. The user can then copy that key value, and set `spec.cleanup.confirmationKey` to the same value to confirm that OLM should proceed with cleanup. More formally OLM applies the following logic:
```

Comment:
can we call this "spec.cleanup.overrideConfirmationKey"  or something else that makes it clearer that it's an override/force field?